### PR TITLE
Frontend setup changes and header component 

### PR DESCRIPTION
**Pull request by lin8231**
* BrowserRouter is now in Index.js to make App.js cleaner 
* Changed default font to Open Sans (should be Google Sans, can't get local fonts working atm - to amend later if time allows)
* Changed logo image to be of correct colour and font 
* HeaderBar component implemented: 
    * this receives a prop which determines whether the inner navigation links are shown (not shown if not logged in or selecting or creating comp
    * should receive a prop which determines whether "logged in" buttons are shown or "logged out" button are shown.
    * Currently only shows "logged out" buttons. 
* LinkButton component implemented. 
    * This is a button that acts as a react router Link. 
    * Has to= and can receive onClick handlers. 
    * Receives a boolean prop which determines whether the colour is blue or white (style). 
* Implemented LoginButtonSet component. 
    * This is a grouping of the "What is sGonks?" and "Sign in" buttons on the home page 
    * Currently used in the header only
    * Made as a component so it can be reused on the login page content itself and also to apply styles easier 
* Linter for js and css files have been removed. Linting for React will be done locally with a local formatter instead.

### CODE REVIEW

File: project/frontend/src/App.js [hunk 1]
```diff
@@ -1,26 +1,29 @@
...
 import CreateCompetition from './containers/CreateCompetition/CreateCompetition'
 import Explanation from './containers/Explanation/Explanation'
 import Login from './containers/Login/Login'
 import SelectCompetition from './containers/SelectCompetition/SelectCompetition'
 import SGonksPlatfrom from './containers/SGonksPlatform/SGonksPlatform'
 
+import HeaderBar from "./components/HeaderBar/HeaderBar";
+
 function App() {
+
```
Comment (roonil-wazlib) [inline project/frontend/src/App.js:15]: Is there a reason for all this blank space?

**Review by roonil-wazlib — CHANGES_REQUESTED** [review]

**Comment by roonil-wazlib** [comment]
Linter is unhappy, can you fix that?

File: project/frontend/src/App.js (hunk 1 above)
Comment (lin8231) [reply project/frontend/src/App.js:15]: This has been fixed in the next PR :)

**Comment by lin8231** [comment]
> Linter is unhappy, can you fix that?

Killed linter.

**Comment by roonil-wazlib** [comment]
@googlebot I consent.

**Review by Phoebe-Khokgawe — APPROVED** [review]

**Review by roonil-wazlib — APPROVED** [review]

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -12,18 +12,6 @@ jobs:
...
-        run: node_modules/css-validator/bin/css-validator project/frontend/src/*.css
-      - name: Validate JavaScript
-        if: always()
-        run: node_modules/eslint/bin/eslint.js project/frontend/src/*.js
-      - name: Check CSS Formatting
-        if: always()
-        run: node_modules/prettier/bin-prettier.js -c project/frontend/src/*.css
-      - name: Check JavaScript Formatting
-        if: always()
-        run: diff -u <(cat project/frontend/src/*.js) <(node_modules/clang-format/bin/linux_x64/clang-format --style=Google project/frontend/src/*.js)
```
Comment (leonnemets) [inline .github/workflows/lint.yml:26]: Can you comment this out instead, and leave a TODO or an explanation of why it is commented out.

I guess you removed it from a view places, so a TODO everywhere may become messing. Leaving it up to you

File: project/frontend/src/components/HeaderBar/HeaderBar.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import LoginButtonSet from "../UI/LoginButtonSet/LoginButtonSet";
+
+const HeaderBar = (props) => {
+  let innerNavLinks = [
+    { linkTo: "/", display: "My sGonks" },
+    { linkTo: "/", display: "Competition" },
+    { linkTo: "/", display: "Marketplace" },
+  ]
+
+  let innerNav = props.innerNav ? (
```
Comment (leonnemets) [inline project/frontend/src/components/HeaderBar/HeaderBar.js:15]: when would props.innerNav  be null?

File: project/frontend/src/index.js [hunk 4]
```diff
@@ -1,19 +1,22 @@
...
-import React from 'react';
-import ReactDOM from 'react-dom';
+import React from "react"
+import ReactDOM from "react-dom"
+import {BrowserRouter} from "react-router-dom";
 
-import App from './App';
-import reportWebVitals from './reportWebVitals';
+import App from "./App"
+import reportWebVitals from "./reportWebVitals"
```
Comment (leonnemets) [inline project/frontend/src/index.js:8]: sometimes a semicolon, sometimes not, do they not matter?

**Review by leonnemets — APPROVED** [review]
LGTM.

I would love start seeing some tests even if you start small. May be you can test that the routing functionality works and takes you to the right place OR you can add some screenshot tests and have golden files of what the UI looks like

File: project/frontend/src/index.js (hunk 4 above)
Comment (lin8231) [reply project/frontend/src/index.js:8]: Sorry this was in between the linter swap, will fix in next PR!

File: project/frontend/src/components/HeaderBar/HeaderBar.js (hunk 3 above)
Comment (lin8231) [reply project/frontend/src/components/HeaderBar/HeaderBar.js:15]: * When the user is not logged in
* When the user is logged in but hasn't selected a competition
Inner nav is for navigation WITHIN a specific competition, so doesn't make sense to have them in the above ^ scenarios.

File: .github/workflows/lint.yml (hunk 2 above)
Comment (lin8231) [reply .github/workflows/lint.yml:26]: @leonnemets Will add a comment in the merge of this PR explaining why it was removed. Would that be fine?

File: project/frontend/src/components/HeaderBar/HeaderBar.js (hunk 3 above)
Comment (lin8231) [reply project/frontend/src/components/HeaderBar/HeaderBar.js:15]: Sorry, just to clairfy. innerNav and props.innerNav are different things. 
props.innerNav is a boolean variable passed into this class.
innerNav is the display of the actual inner navigation items. Which would be empty (null) or populated. 

innerNav displays things if props.innerNav is true.

Comment (lin8231) [reply project/frontend/src/components/HeaderBar/HeaderBar.js:15]: Hopefully that clears things up :)

**Pull request merged**